app/components/c-add-to-cart.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div :class="b()">
+    <div :class="b('quantity')">
+      <e-input
+        v-model.number="quantity"
+        :min="step"
+        :step="step"
+        type="number"
+        name="quantity"
+      />
+    </div>
+    <div :class="b('button')">
+      <e-button
+        :progress="progress"
+        primary
+        width="full"
+        @click="onClick"
+      >
+        <e-icon :class="b('icon')" icon="i-cart" inline/><span v-if="hasLabel"> {{ $t('c-add-to-cart.addToCart') }}</span>
+      </e-button>
+    </div>
+    <!-- TODO - remove -->
+    <div ref="debug">{{ debug }}</div>
+  </div>
+</template>
+
+<script>
+  import { mapActions } from 'vuex';
+
+  export default {
+    name: 'c-add-to-cart',
+    // components: {},
+    // mixins: [],
+
+    props: {
+      /**
+       * Sku of the article to be added
+       */
+      sku: {
+        type: String,
+        required: true,
+      },
+
+      /**
+       * Increment steps. Also defines initial value
+       */
+      step: {
+        type: Number,
+        default: 1,
+      },
+
+      /**
+       * Show label or not
+       */
+      label: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    data() {
+      return {
+        quantity: this.$props.step,
+        hasLabel: !!this.$props.label,
+        progress: false,
+        debug: '', // TODO - remove
+      };
+    },
+
+    // computed: {},
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      ...mapActions({
+        addToCart: 'cart/addToCart',
+      }),
+      onClick() {
+        this.progress = true;
+
+        this.addToCart(this.sku, this.quantity)
+          .then((response) => {
+            this.progress = false;
```

Comment:
- [ ] Please use `.finally()` for data resets after a promise.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
Finally doesn't receive a parameter which i need for debug output. I have to refactor this anyway in WF-600-mini-cart

Reply:
@geroldboehler Handling the response in `then()` is ok. But `this.progress = false` should be handled inside `finally()` to make sure it's always executed.